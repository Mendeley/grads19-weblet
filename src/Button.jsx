import React from "react";
import styled from "styled-components";

export const StyledButton = styled.button`
  text-align: center;
  background: lightgrey;
`;

const Button = ({ children, onClick, ...props }) => {
  return (
    <StyledButton onClick={onClick} {...props}>
      {children}
    </StyledButton>
  );
};

export default Button;
