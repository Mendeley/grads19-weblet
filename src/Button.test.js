import Button, { StyledButton } from "./Button";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import React from "react";

configure({ adapter: new Adapter() });

describe("Button", () => {
  it("renders a blank button", () => {
    expect.assertions(1);

    const result = shallow(<Button />);

    expect(result.debug()).toMatchSnapshot();
  });
  it("renders a button with text", () => {
    expect.assertions(1);

    const result = shallow(<Button children="Test" />);

    expect(result.debug()).toMatchSnapshot();
  });
  it("executes passed function when clicked", () => {
    expect.assertions(1);

    const callBack = jest.fn();

    const wrapper = shallow(<Button onClick={callBack} children="Test" />);

    wrapper.find(StyledButton).simulate("click");

    expect(callBack).toHaveBeenCalledTimes(1);
  });
});
