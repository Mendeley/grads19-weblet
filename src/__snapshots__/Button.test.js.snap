// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders a blank button 1`] = `"<styled.button onClick={[undefined]} />"`;

exports[`Button renders a button with text 1`] = `
"<styled.button onClick={[undefined]}>
  Test
</styled.button>"
`;
